state/state.go

```diff
@@ -335,11 +335,6 @@ func MakeGenesisDocFromFile(genDocFile string) (*types.GenesisDoc, error) {
 
 // MakeGenesisState creates state from types.GenesisDoc.
 func MakeGenesisState(genDoc *types.GenesisDoc) (State, error) {
-	err := genDoc.ValidateAndComplete()
```

Comment:
I'd instead leave this because, if the state is partially gone (due to FS corruption or other reasons; `stateKey` is gone, but `genesisDocKey` is there, although the probability is very low), `LoadFromDBOrGenesisDoc` makes sure we always validate `genDoc`.

Reply:
Yes why not. Two rights make a right.